### PR TITLE
Navigation warnings

### PR DESCRIPTION
**Pull request by calellowitz**
This PR changes warning messages for unsaved cloudcare pages to match the language in the dialog boxes and adds those dialog boxes when changing forms in addition to just navigating away from the page.

Original ticket here
http://manage.dimagi.com/default.asp?108996#1016215
@nickpell

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/shared.js [hunk 1]
```diff
@@ -34,7 +34,14 @@ var Selectable = Backbone.View.extend({
                 this.trigger("deselected");
             }
         } else {
-            this.select();
+            if (window.mainView.router.view.dirty) {
+                var dialog = confirm(translatedStrings.sidebarDirty);
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/shared.js]: 'dialog' is already defined.

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/shared.js [hunk 2]
```diff
@@ -34,7 +34,14 @@ var Selectable = Backbone.View.extend({
...
+                if (dialog == true) {
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/shared.js]: Use '===' to compare with 'true'.

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/shared.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ var Selectable = Backbone.View.extend({
         }
         if (this.selected) {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/shared.js]: i'm thinking this could benefit from a slight refactor since the logic is getting a bit hard to follow. we can substantially cut down the lines here by pulling the confirmation and dirty check above the `if (this.selected)`

**Comment by calellowitz** [comment]
@benrudolph should be fixed. thanks for the help

**Comment by benrudolph** [comment]
nice i like it
